hameed_hw/index.js

```diff
@@ -0,0 +1,89 @@
+// Create an http server that can add and subtract from a number, which we will call the "state".
+// To run this file: node .
+// Then in your browser: http://localhost:3000
+let http = require('http');
+// declare the state variabel which start from 10
+let state = 10
+let server = http.createServer();
+// give the port a number
+const port = 3000;
+//To listen to the port and console the state of this port 
+server.listen(port, function(error) {
+  if (error) {
+    console.log(error);
+  } else {
+    console.log('api listening on port', port);
+  }
+});
+// creat a response for the request url
+server.on('request', function(request, response) {
+  //console.log(response);
+  console.log('New http request received', request.url);
+  //use switch to determine the request url case and prosses the resultes
+  switch (request.url) {
+  	// /state 
+	// response: the current state in a html format 
+	// when the server starts, this should return "10"
+  	//in this case we just show the curent state value 
```

Comment:
I love how you put the spec right in the comments. People refer to this as having the documentation "close to the code". It's a great habit to get into.